src/lib.rs

```diff
@@ -42,10 +43,29 @@ use std::{
     os::unix::fs::PermissionsExt,
     path::{Path, PathBuf},
     process::{Command, Stdio},
-    time::{Duration, Instant},
+    time::Duration,
 };
 use tar::Archive;
 
+/// How or where to build the binaries from.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum SnCodebaseType {
+    Main {
```

Comment:
It might be worth putting some doc comments on this entry just to say that if features are used, the binary will be built, but otherwise the 'latest' will be fetched from S3.